### PR TITLE
Feature/GHI-11-freestyle-jobs & bug/GHI-7-Remove-dummy-declarative-methods

### PR DESCRIPTION
**Pull request by callbacksin**
* implemented DowloadDatasetStep method via a freestyle job
* fix the bug showing dummy methods in a build steps list
* add minor fixes

### CODE REVIEW

**Comment by callbacksin** [comment]
related: #11 , #7

**Review by VKrus — APPROVED** [review]

File: src/main/kotlin/org/zowe/zdevops/logic/DownloadOperation.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/*
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/zdevops/logic/DownloadOperation.kt:1]: Tests at least 80% of instructions

File: src/main/kotlin/org/zowe/zdevops/utils/FieldsValidation.kt [hunk 2]
```diff
@@ -27,7 +27,15 @@ fun validateDatasetName(dsn: String): FormValidation? {
...
+    val memberPattern = Regex("^(?:[A-Z#@\$][A-Z0-9#@\$]{0,7}|[a-z#@\$][a-zA-Z0-9#@\$]{0,7})\$")
+
+    return if (member.length > 8 || member.isEmpty()) {
+        FormValidation.error("The field must be between 1 and 8 characters in length")
+    } else if(!member.matches(memberPattern)) {
+        FormValidation.warning("It seems the member name is invalid")
+    } else {
+        FormValidation.ok()
+    }
 }
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/zdevops/utils/FieldsValidation.kt:39]: Tests

File: src/main/kotlin/org/zowe/zdevops/utils/miscUtils.kt [hunk 3]
```diff
@@ -1,32 +1,2 @@
 package org.zowe.zdevops.utils
```
Comment (KUGDev) [inline src/main/kotlin/org/zowe/zdevops/utils/miscUtils.kt:1]: Remove as not needed anymore

**Comment by KUGDev** [comment]
classic/steps/DeleteDatasetStep.kt - 80% coverage needed
classic/steps/DeleteDatasetsByMaskStep.kt - 80% coverage needed
logic/DeleteOperation.kt - 80% coverage needed

**Review by KUGDev — CHANGES_REQUESTED** [review]
classic/steps/DeleteDatasetStep.kt - 80% coverage needed
classic/steps/DeleteDatasetsByMaskStep.kt - 80% coverage needed
logic/DeleteOperation.kt - 80% coverage needed

**Review by KUGDev — APPROVED** [review]

**Pull request merged**